### DISCUSSION TITLE
OpenAPI examples

### DISCUSSION THREAD

**Discussion by aknuds1** [cuelang/cue#897] [Q&A] [answered]
Do there exist any examples of generating OpenAPI from service schema definitions in CUE? Apologies if such examples are in plain sight, but I've had little success in finding them.

**Comment by PierreR** [comment]
FWIW I would be interested in generating CUE definitions from an OpenAPI json file. Maybe both concerns can be answered in one shot ;-)

**Comment by mpvl** [comment, marked as answer]
For an elaborate example of Proto -> CUE -> OpenAPI pipeline (API usage), see https://github.com/istio/tools/tree/master/cmd/cue-gen.

For command line examples of CUE -> OpenAPI, see `cue help filetypes`.

**Reply by PierreR** [reply]
The (inverse) convertion OpenAPI -> CUE is not currently supported. Is there a reason why such convertion is less of a priority ?

**Reply by myitcv** [reply]
Simply that there hasn't been sufficient demand, compared to other priorities. 

One thing that crossed my mind as I created https://github.com/cuelang/cue/issues/1012 however is that we should create placeholder issues for the "missing" conversions. That way, people have a clear place to vote, add context about priorities, timelines etc. Would that work for you, @PierreR?

**Reply by PierreR** [reply]
@myitcv Sure. FFIW the conversion from OpenAPI -> CUE is not a blocking point as far as I am concerned because I am mainly interested in importing Kubernetes schemas at this point (which are written in Go). That said, intuitively it seems that such conversion would make the CUE ecosystem less "bound" to Go in general.

**Reply by myitcv** [reply]
> That said, intuitively it seems that such conversion would make the CUE ecosystem less "bound" to Go in general.

Indeed. Just to be clear, all conversions are very much on the table! Even for other languages. It's just a case of some necessarily being higher priority than others, and other parts of CUE necessarily being higher priority than some conversions: a function of time/people available! My point was that our priorities are largely determined by users' interest/priorities, so creating issues for the "missing" conversions would be a way of providing a point to gather interest/details. I'll create a couple such issues to see where they take us.

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/897.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
